### PR TITLE
Fix deletion of unprotected branches

### PR DESCRIPTION
**Pull request by daviian**
Targets #2629 

PR only targets release v1.2 as it is already fixed on master

### CODE REVIEW

**Review by jonasfranz — APPROVED** [review]

**Comment by lunny** [comment]
LGTM

**Review by lafriks — CHANGES_REQUESTED** [review]
But for unprotected `protectBranch` should be nil & `CanPush` should be used only for pushing code check not for branch deletion

**Comment by daviian** [comment]
@lafriks I can change internal api too. At the moment the internal api returns a protectedBranch with canPush: true if there is none.

**Comment by lafriks** [comment]
I think that would be correct approach as this is also what master branch does

**Comment by daviian** [comment]
@lafriks done

**Comment by lafriks** [comment]
@daviian please fix go fmt error

**Comment by lafriks** [comment]
LGTM

**Review by lafriks — APPROVED** [review]

File: models/branches.go [hunk 1]
```diff
@@ -38,6 +38,11 @@ func (protectBranch *ProtectedBranch) BeforeUpdate() {
 	protectBranch.UpdatedUnix = time.Now().Unix()
 }
 
+// IsProtected returns if the branch is protected
+func (protectBranch *ProtectedBranch) IsProtected() bool {
+	return protectBranch.ID > 0
```
Comment (lafriks) [inline models/branches.go:43]: @daviian you can change here to `protectBranch != nil && protectBranch.ID > 0` and than you can remove that nil check above

Comment (daviian) [reply models/branches.go:43]: @lafriks Current implementation would reflect current master. But if you want, of course I can add the check inside the function.

Comment (lafriks) [reply models/branches.go:43]: @daviian ok, let it be this way than for now, this could be optimized in master than

**Comment by lafriks** [comment]
LGTM

**Pull request merged**